// pages/createArticle/createArticle.js
import React, { useEffect, useState, useContext } from 'react';
import ArticleForm from 'components/articleForm'
import useFetch from 'hooks/useFetch'
import { Redirect } from 'react-router-dom'
import {CurrentUserContext} from 'contexts/currentUser'

const CreateArticle = () => {
  const _apiUrl = '/articles/'
  const [{  response, error}, doFetch] = useFetch(_apiUrl)
  const [isSuccessfullSubmit, setIsSuccessfullSubmit] = useState(false)
  const [currentUserState] = useContext(CurrentUserContext)
  const initialValues = {
    title: '',
    description: '',
    body: '',
    tagList: []
  }

  const handleSubmit = article => {
    // console.log('hande submit', article)
    doFetch({
      method: 'POST',
      data: {
        article
      }
    })
  }

  useEffect(() => {
    if (!response) {
      return
    }
    setIsSuccessfullSubmit(true)

  }, [response, isSuccessfullSubmit])

  if (currentUserState.isLoggedIn === false) {
    return <Redirect to='/'/>
  }

  if (isSuccessfullSubmit) {
    return <Redirect to={`/articles/${response.article.slug}`}/>
  }

  return (
    <div>
      <ArticleForm
      errors={ (error && error.errors) || {} }
      onSubmit={handleSubmit}
      initialValues={initialValues}/>
    </div>
  );
};

export default CreateArticle;
